protobufs/livekit_rtc_v2.proto

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package livekit;
+option go_package = "github.com/livekit/protocol/livekit";
+option csharp_namespace = "LiveKit.Proto";
+option ruby_package = "LiveKit::Proto";
+
+import "livekit_rtc.proto";
+import "livekit_models.proto";
+
+message Signalv2ClientMessage {
+  // monotonically increasing number (by 1) starting at some non-zero number (should not rollover)
+  uint32 message_id = 1;
+
+  // last remote message processed
+  uint32 last_processed_remote_message_id = 2;
+
+  oneof message {
+    Fragment fragment = 3;
+    Signalv2ClientEnvelope envelope = 4;
+    ConnectRequest connect_request = 5;
+  }
+}
+
+message Signalv2ClientEnvelope {
+  // collection of client -> server messages
+  repeated Signalv2ClientMessage client_messages = 1;
+}
+
+message Signalv2ServerMessage {
+  // monotonically increasing number (by 1) starting at some non-zero number (should not rollover)
+  uint32 message_id = 1;
+
+  // last remote message processed
+  uint32 last_processed_remote_message_id = 2;
+
+  repeated ErrorResponse errors = 3;
+
+  oneof message {
+    Fragment fragment = 4;
+    Signalv2ServerEnvelope envelope = 5;
+    ConnectResponse connect_response = 6;
+  }
+}
+
+message Signalv2ServerEnvelope {
+  // collection of server -> client messages
+  repeated Signalv2ServerMessage server_messages = 1;
+}
+
+message ConnectionSettings {
+  bool auto_subscribe = 1;
+  bool adaptive_stream = 2;
+  optional bool subscriber_allow_pause = 3;
+  bool disable_ice_lite = 4;;
+}
+
+message ConnectRequest {
+  ClientInfo client_info = 1;
+  ConnectionSettings connection_settings = 2;
+  map<string, string> participant_attributes = 3;
```

Comment:
are we consciously deciding to allow the connection request to only include attributes and not metadata? 

How would we handle a connection request that tries to set attributes but the permission is missing? It would be weird to deny the connection, but it would still be good to inform clients about that part of the request not being successful.

Reply:
oh, and also, how would we handle conflicting entries in attributes that are stored in the token vs in the request here?

when you say clients would get a 401 response in case `canUpdateOwnMetadata` is false, would that in turn mean no connection can get established?
Would that differ from a user's token not being valid?

Reply:
This [commit ](https://github.com/livekit/protocol/pull/1094/commits/e27eb0acd7330b9aae2cc04e289ccc5a325ae4bd)has the changes. Please let me know if this looks okay.

Reply:
I read the code wrong. It actually replaces key value if there is a key collision and the value is not empty value. Sorry, will update the comment.

In that light, I am thinking if `metadata` is provided here, it should replace the one in the token?

Reply:
I think consistency is good here, so mirroring what we do for attributes makes sense to me

Reply:
Sounds good. Will update comments and commit. Will also change metadata overwrite behaviour in server.